pkg/controller/job/job_controller.go

```diff
@@ -771,10 +789,12 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 		return err
 	}
 
+	syncJobContext.pods = pods
 	activePods := controller.FilterActivePods(pods)
```

Comment:
I think we should get rid of these independent variables and only use the ones from the `syncJobCtx`, for consistency.

Reply:
Done